feat(fields): Consolidate field definitions and introduce rich descriptions

jennmueng opened the pull request:
This is the feature branch for the work on consolidating the field definitions into one single place. This affects a ton of features throughout the app but we're mainly concerned about the smart search bar.

Includes new rich definitions provided by @doralchan and also the refactorings required to consolidate these definitions.

Each of these PRs have already been individually reviewed and merged into this feature branch.

jennmueng committed:
ref(fields): Move field definitions to single file for issues view (#36276)

Moves types and provides definitions for the fields used in the searchBar into a single file.

There will be a PR following this one refactoring the field definitions for other uses such as discover. These are separate to reduce PR size for reviewability.

There are some definitions missing, but this PR's scope is to just ensure the definitions on the Issues screen searchbar is correct. There are also valid fields that don't show up in the issues searchbar such as age but these cases will be handled again in another PR due to fear of it becoming hard to review

evanpurkhiser commented on static/app/utils/fields/index.ts:
```diff
@@ -0,0 +1,574 @@
+
+export const FIELDS: Record<string, FieldDefinition> = {
+  [FieldKey.AGE]: {
+    desc: 'The age of the issue in relative time',
```
Should all of the `desc`'s be wrapped in `t()`

jennmueng replied:
we don't do any translations currently right, just in case for the future?

evanpurkhiser replied:
so when you wrap it in `t()` it will get included in strings that can be translated. So next time we do translations this will be in the set of strings to translate

evanpurkhiser commented on static/app/utils/fields/index.ts:
```diff
@@ -0,0 +1,574 @@
+  desc?: string;
+}
+
+export const FIELDS: Record<string, FieldDefinition> = {
```
Should it be
```suggestion
export const FIELDS: Record<FieldKey, FieldDefinition> = {
```

jennmueng committed:
ref: ColumnType uses values from FieldValueType instead of literals (#36873)

Refactor ColumnType to use values from FieldValueType this is a stopgap instead of using the enum to avoid having to completely refactor the tons of times ColumnType is used.

jennmueng committed:
ref(fields): Refactor discover/alerts fields to single unified fields definition (#36340)

Refactors discover/alerts/dashboard fields to use the new fields definitions.

Also adds definitions for the Web/Mobile Vitals and refactor the discover fields to use these new definitions

jennmueng committed:
fix(issues-fields): Add missing issue field keys (#37091)

The list of valid issue keys was originally copied from the tagStore. They were missing a lot of keys defined in https://docs.sentry.io/product/sentry-basics/search/searchable-properties/ as available for issues such as `age` and at the same time I missed the fields from `TagStore.getIssueAttributes`:

https://github.com/getsentry/sentry/blob/bdaabf26c08e6f1ca7a1578fa01525156086f9df/static/app/stores/tagStore.tsx#L56-L120

This PR adds these missing fields in.

jennmueng committed:
ref(fields): Wrap all descriptions with a t (#37089)

`t` everything

jennmueng committed:
Update static/app/utils/fields/index.ts

Co-authored-by: Evan Purkhiser <evanpurkhiser@gmail.com>

jennmueng committed:
fix: & the field keys instead of |

jennmueng committed:
ref(tagStore): Cleanup tagstore and refactor to make more sense (#37139)

`getAllTags` never made sense because it was never all the tags, just the state tags, use `getStateTags` instead. `getBuiltInTags` was also removed and functionality moved to `getIssueTags`

jennmueng committed:
feat(fields): Definitions and refactoring for aggregation fields (#36386)

Define definitions for aggregation functions + refactor uses of them to use new definitions from the consolidated field definitions.

Old string literal union type `AggregationKey` that contains aliases is renamed to `AggregationKeyWithAlias` while the new `AggregationKey` is an enum.

Co-authored-by: William Mak <william@wmak.io>

evanpurkhiser commented:
Looks like some tests are failing

jennmueng committed:
fix: Fix invalid import after rebase
